### PR TITLE
Extract selfExecute in SDK

### PR DESCRIPTION
**Pull request by marekkirejczyk**

### CODE REVIEW

File: universal-login-sdk/lib/sdk.ts [hunk 1]
```diff
@@ -60,78 +60,21 @@ class UniversalLoginSDK {
...
-    const data = new utils.Interface(WalletContract.interface).functions.addKey.encode([key, keyPurpose]);
-    const message = {
-      ...transactionDetails,
-      to,
-      from: to,
-      data,
-    };
-    return this.execute(message, privateKey);
+  async addKey(to: string, publicKey: string, privateKey: string, transactionDetails: Message, keyPurpose = MANAGEMENT_KEY) {
+    return this.selfExecute(to, 'addKey', [publicKey, keyPurpose], privateKey, transactionDetails);
```
Comment (gasiortomasz) [inline universal-login-sdk/lib/sdk.ts]: Just to be consistent with addKeys(): keyPurpose -> keyRole?

Comment (gasiortomasz) [reply universal-login-sdk/lib/sdk.ts]: `return this.selfExecute(to, 'addKeys', [[publicKey], [keyRole]], privateKey, transactionDetails);`

File: universal-login-sdk/test/SdkSigner.ts [hunk 2]
```diff
@@ -4,11 +4,13 @@ import {loadFixture} from 'ethereum-waffle';
 import basicWalletService from './fixtures/basicSDK';
 import {SdkSigner} from '../lib/SdkSigner';
 
-describe('SdkSigner', () => {
+describe('E2E: SdkSigner', () => {
```
Comment (gasiortomasz) [inline universal-login-sdk/test/SdkSigner.ts]: Should we rename `SdkSigner -> SDKSigner` ?

File: universal-login-sdk/lib/sdk.ts [hunk 3]
```diff
@@ -210,31 +163,20 @@ class UniversalLoginSDK {
   }
 
   async denyRequest(walletContractAddress: string, publicKey: string, privateKey: string) {
-    const cancelAuthorisationRequest: CancelAuthorisationRequest = {
-      walletContractAddress,
-      publicKey,
-      signature: ''
-    };
+    const cancelAuthorisationRequest = { walletContractAddress, publicKey};
```
Comment (xitronix) [inline universal-login-sdk/lib/sdk.ts]: Formatting (Delete space)

Comment (marekkirejczyk) [reply universal-login-sdk/lib/sdk.ts]: ✅

File: universal-login-sdk/test/SdkSigner.ts (hunk 2 above)
Comment (marekkirejczyk) [reply universal-login-sdk/test/SdkSigner.ts]: Good question, not sure

**Pull request merged**